torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -229,6 +249,9 @@ def parallelize_llama(model, world_mesh, parallel_dims, job_config: JobConfig):
                 reshard_after_forward=reshard_after_forward,
             )
             model.layers[layer_id] = transformer_block
+        if enable_compile_rmsnorm:
+            model.norm = torch.compile(model.norm, dynamic=False)
```

Comment:
I think we should also check the fqn, iirc `torch.compile` by default would prepend a fqn for the compiled model, we should try to get rid of it in some way, otherwise we would not be able to load Llama3 pretrained weights

Reply:
you are right - it goes from:
~~~
(attention_norm): RMSNorm()
(ffn_norm): RMSNorm()
~~~
to 
~~~
(attention_norm): OptimizedModule(
 (_orig_mod): RMSNorm() )
 (ffn_norm): OptimizedModule(
(_orig_mod): RMSNorm()
~~~

let me clean this up.

Reply:
will make a cleanup function

Reply:
IMO we should probably not make a state_dict hook if that's what u are going to do.

Instead we should compile the norm function directly

Reply:
distributed_state_dict already has this covered. It will remove the FQN from torch.compile. Let me know if you do see some issues.

Reply:
An alternative, if you are still seeing this error, is to call `module.compile` instead of `torch.compile`